Library/Homebrew/github_packages.rb

```diff
@@ -421,13 +426,28 @@ def upload_bottle(user, token, skopeo, formula_full_name, bottle_hash, keep_old:
       validate_schema!(IMAGE_MANIFEST_SCHEMA_URI, image_manifest)
       manifest_json_sha256, manifest_json_size = write_hash(blobs, image_manifest)
 
-      {
+      bottle_manifests = T.let([{
         mediaType:   "application/vnd.oci.image.manifest.v1+json",
         digest:      "sha256:#{manifest_json_sha256}",
         size:        manifest_json_size,
         platform:    platform_hash,
         annotations: descriptor_annotations_hash,
-      }
+      }], T::Array[T::Hash[Symbol, String]])
+
+      if (all_files = tag_hash["all_files"].presence)
+        all_files_json_sha256, all_files_size = write_hash(blobs, all_files)
+        bottle_manifests << {
+          mediaType:   "application/sh.brew.all_files+json",
```

Comment:
This is valid under image-spec 1.1.0:

> An encountered `mediaType` that is unknown to the implementation MUST NOT generate an error.

[`skopeo` does not follow the spec yet](https://redirect.github.com/containers/image/issues/2030#issuecomment-1656321976), however this is likely a misuse of manifests anyway. What you probably want instead is actually a manifest with contents that look something like point 2 in https://github.com/opencontainers/image-spec/blob/main/manifest.md#guidelines-for-artifact-usage.

Reply:
Ok, thanks for the pointers @Bo98. I've run out of time I can justify spending on the `all_files` stuff here but I've opened https://github.com/Homebrew/brew/pull/17535 and assigned to you if you want to finish it off (feel free to commit or force push directly to that branch and approve and merge when you're ready). If you'd rather also bail: feel free to let me know and/or just close that PR out.